ViewModels/PanZoomViewModel.cs

```diff
@@ -68,6 +68,33 @@ public bool IsPanning
         private Data.RectRegion _currently_creating_region;
         private System.Nullable<Point> _last_pan_position;
 
+        /// <summary>
+        /// adds a filled region.
+        /// </summary>
+        public void FillRegion()
+        {
+            if (InputPack.SelectedRegionBrush.SelectedEmulatedDevice == null || InputPack.SelectedRegionBrush.SelectedEmulatedKey == null)
+            {
+                return;
+            }
+
+            Data.RectRegion f = new Data.RectRegion() { ScaleFactor = InputPack.EditingTexture.ScaleFactor, X = 0, Y = 0, Height = InputPack.EditingTexture.ImageHeight, Width = InputPack.EditingTexture.ImageWidth, Device = InputPack.SelectedRegionBrush.SelectedEmulatedDevice, Key = InputPack.SelectedRegionBrush.SelectedEmulatedKey, OwnedTexture = InputPack.EditingTexture };
+
+            //one fill region is enough per key
+            foreach (Data.RectRegion r in InputPack.EditingTexture.Regions)
```

Comment:
For the same reason we don't currently handle overlapping regions, maybe we want to avoid creating a fill region if there are already _any_ regions existing.

Reply:
now it is looked whether a filled region is already present, this is then adapted.

Reply:
That's better but there's another case.  We need to handle the case users have placed _other_ regions and then choose to fill.  This isn't your fault, but probably worthwhile to fix this while we're at it.

We have two options:

1)  Abort the fill if another region exists:

```
if (InputPack.EditingTexture.Regions.Count > 0)
{
	foreach (Data.RectRegion r in InputPack.EditingTexture.Regions)
	{
		if (r.X == f.X & r.Y == f.Y & r.Width == f.Width & r.Height == f.Height)
		{
			r.Device = f.Device;
			r.Key = f.Key;
		}
		else
		{
			// Existing regions already exist abort
			return;
		}
	}

	return;
}


InputPack.EditingTexture.Regions.Add(f);
```

2)  Clear the existing regions and add the fill.

```
if (InputPack.EditingTexture.Regions.Count > 0)
{
	foreach (Data.RectRegion r in InputPack.EditingTexture.Regions)
	{
		if (r.X == f.X & r.Y == f.Y & r.Width == f.Width & r.Height == f.Height)
		{
			r.Device = f.Device;
			r.Key = f.Key;
		}
	}

	return;
}


InputPack.EditingTexture.Regions.Clear();
InputPack.EditingTexture.Regions.Add(f);
```

I think 2 seems a bit more natural

Reply:
> 1. Abort the fill if another region exists:
Is already present
```
if (InputPack.EditingTexture.Regions.Count > 0)
{
	foreach (Data.RectRegion r in InputPack.EditingTexture.Regions)
	{
		if (r.X == f.X & r.Y == f.Y & r.Width == f.Width & r.Height == f.Height)
		{
			r.Device = f.Device;
			r.Key = f.Key;
		}
	}

	return; <---------- here
}
```

Reply:
Mmmh, you are right.  If we are going to abort, we need to probably let the user know.  Possibly with a messagebox?

Reply:
I would like to do another time, after the import function I will clean up a bit, that should make things easier.